docs/intro/index.md

```diff
@@ -4,25 +4,92 @@
 
 This is a work in progress.
 
+We'd love to hear your feedback! If you find any errors or have suggestions for
+the book, please [file an
+issue](https://github.com/ahrefs/melange-for-react-devs/issues) or ping @feihong
+on the [#melange channel in the Reason Discord](https://discord.gg/reasonml).
+
 :::
 
-## Motivation
+## What is Melange?
+
+Melange is a collection of tools that bring the robustness of a mature,
+statically-typed multi-paradigm programming language to the JavaScript
+ecosystem. That language is OCaml, a language invented in the 1990s which has
+been battle-tested by industry stalwarts. The heart of Melange is a compiler
+that translates OCaml to human-readable JavaScript and built-in language
+constructs for zero-cost interoperation with JavaScript.
+
+## Why OCaml?
+
+OCaml codebases scale well both in terms of quantity of lines and number of
+contributors. The sound type system helps to prevent ambiguous behavior in your
+program---if it compiles, it runs. OCaml and React (via
```

Comment:
The "if it compiles, it runs" is a bit confusing because TypeScript also "runs if it compiles" 😄 What about 
```suggestion
program---if it compiles, it runs without runtime errors. OCaml and React (via
```
?